infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -567,6 +580,30 @@ private void handleTxToken(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerCont
         handleAllocation(msg, ctx, r);
     }
 
+    /**
+     * this method serves a monotonically increasing counter as an Ordered GUID.
+     * It simply returns a counter along with the current epoch value.
+     * If the sequencer restarts the epoch ticks up while the counter resets, thereby
+     * allowing a globally unique monotonically increasing sequence to be served.
+     *
+     * Not Thread Safe: Only OK when SequencerServer is single threaded.
+     *
+     * @param msg corfu message containing guid request
+     * @param ctx netty ChannelHandlerContext
+     * @param r   server router
+     */
+    private void handleOrderedGuidToken(CorfuPayloadMsg<TokenRequest> msg,
```

Comment:
Since ordering is only determined post commit, this id generation scheme should only address the "uniqueness" issue, where ids don't collide, ordering is not required at this point.

You acquire a token once (during initializing), not per enqueue and then use that to generate unique tokens within that client.

Reply:
Sure, but here the ordering comes for free, so it's just advertised as such. Any issue with advertising the ordering?

Reply:
True, but now we don't even need the sequencer to generate the GUID because its being generated on the client via the regular token acquire.

Reply:
Got it. Will change this.